### PR TITLE
feat(datatrak): RN-1340: Task cancelling

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Add Save Tasks Route to handle updating a task
- Add Task Actions Menu and action for cancelling
- Add Cancel Modal

---

### Screenshots:
In linear

### CODE REVIEW

File: packages/datatrak-web-server/src/app/createApp.ts [hunk 1]
```diff
@@ -6,47 +6,49 @@
...
-  ProjectsRoute,
-  ProjectsRequest,
-  SurveyRequest,
-  SurveyRoute,
-  SingleEntityRequest,
-  SingleEntityRoute,
+  ActivityFeedRequest,
+  ActivityFeedRoute,
+  EntitiesRequest,
+  EntitiesRoute,
```
Comment (rohan-bes) [inline packages/datatrak-web-server/src/app/createApp.ts:21]: Thanks for sorting these! :pray:

**Review by rohan-bes — APPROVED** [review]
Looks great!! :tada:

File: packages/datatrak-web/src/features/Survey/SurveyLayout.tsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ export const SurveyLayout = () => {
   } = useSurveyForm();
   const { handleSubmit, getValues } = useFormContext();
   const { mutate: submitSurveyResponse, isLoading: isSubmittingSurveyResponse } =
-    useSubmitSurveyResponse();
+    useSubmitSurveyResponse(from);
```
Comment (rohan-bes) [inline packages/datatrak-web/src/features/Survey/SurveyLayout.tsx:90]: Is this for taking you back to the tasks page after 'completing' a task?

Comment (tcaiger) [reply packages/datatrak-web/src/features/Survey/SurveyLayout.tsx:90]: yes that's right

File: packages/datatrak-web-server/src/routes/SaveTaskRoute.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+export class SaveTaskRoute extends Route<SaveTaskRequest> {
+  public async buildResponse() {
+    const { status } = this.req.body;
+    const { taskId } = this.req.params;
+    const { task } = this.req.models;
+
+    // Update task if id exists
+    if (taskId) {
+      await task.updateById(taskId, { status });
```
Comment (tcaiger) [inline packages/datatrak-web-server/src/routes/SaveTaskRoute.ts:27]: Note: need to update to integrate with this [PR](https://github.com/beyondessential/tupaia/pull/5763/files#diff-cc52ec21ed7d955ee22da2eb18bf330290a5c79cf4f09d30f27ca17ae88c5151). Should also use the api class from context to ensure that it does through permission handling.

**Pull request merged**